src/modules/PublicQuestionPage/QuestionForm.tsx

```diff
@@ -80,6 +84,15 @@ export function QuestionForm({ owner }: { owner: UserProfile }) {
     if (process.env.NODE_ENV === 'development') {
       await sendQuestion(owner?.slug || '', data.q, 'development')
     } else {
+      if (isLogin && user?.email?.includes(owner.slug)) {
```

Comment:
Hmmm, actually user can update the slug after first login.
So detecting by matching with the email is misslead.

Reply:
The correct one is by matching the uid.
But for the public page, I already strip the uid from the response for security purposes 😂

Reply:
This feature seems quite hard to solve for now.

Reply:
One approach that I can think is:

- Passing authorization token, if user logged in
- In the server API:
  - Get session by the token
  - Get uid from the session we get (if exist)
  - if the uid is the same with the owner of the question, you can reject it.

Seems complicated and adding at least 1 API call to the session DB

Reply:
why not just matching with the slug?

`src/app/p/[slug]/page.tsx`
in that page, just disabled the form if slug and is match and give some information about cannot ask to himself

compare slug from params and slug from owner.data.slug

Reply:
hi sir, how about comparing slug from browser URL and slug from account info?, we can do this on file `src/modules/PublicQuestionPage/QuestionForm.tsx` ?

when owner of the page click `Kirim Pertanyaan` button, just show a dialog to tell that owner of the page cannot ask to themself

Reply:
Oh yes, it's possible I guess.
Just can we also adding a way to submit question for our self.
In some cases, let say for testing, maybe we need that 🙈

For initial, can we enable the feature on the NODE_ENV === 'production' only? Or adding new env var `NEXT_PUBLIC_ENABLE_SELF_SUBMISSSION`